salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+        """Compare and configure audit devices
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_devices {list} -- audit devices from remote vault server
+            local_devices {list} -- audit devices from local vault config file
+            ret {dict} -- salt state result
+        """
+        log.info('Processing and configuring audit devices...')
+        new_audit_devices = []
+        try:
+            for audit_device in local_devices:
+                log.debug('Checking if audit device "%s" at path "%s" is enabled...',
+                          audit_device.type, audit_device.path)
+
+                if audit_device in remote_devices:
+                    log.debug('Audit device "%s" at path "%s" is already enabled.',
+                              audit_device.type, audit_device.path)
+                else:
+                    log.debug(
+                        'Audit device "%s" at path "%s" is not enabled. Enabling now...',
+                        audit_device.type,
+                        audit_device.path
+                    )
+                    new_audit_devices.append(audit_device.type)
+                    client.sys.enable_audit_device(
+                        device_type=audit_device.type,
+                        path=audit_device.path,
+                        description=audit_device.description,
+                        options=audit_device.config
+                    )
+                    log.debug('Audit device "%s" at path "%s" is enabled.',
+                              audit_device.type, audit_device.path)
+
+            # Build return object
+            ret['changes']['old'] = [ob.type for ob in remote_devices]
+            ret['changes']['new'] = new_audit_devices or "No changes"
+        except Exception:
+            raise
+
+        log.info('Finished processing audit devices.')
+
+    def cleanup_audit_devices(self, client, remote_devices, local_devices, ret):
+        """Disabling any audit devices not present in the local config file
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_devices {list} -- list of remote audit devices
+            local_devices {list} -- list of local audit devices
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up audit devices...')
+        has_changes = False
+        try:
+            for audit_device in remote_devices:
+                if audit_device not in local_devices:
+                    log.info('Disabling audit device "%s" at path "%s"...',
+                            audit_device.type, audit_device.path)
+                    has_changes = True
+                    client.sys.disable_audit_device(
+                        path=audit_device.path
+                    )
+
+            if has_changes:
+              ret['changes']['new'] = [ob.type for ob in local_devices]
+
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.